workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/DialogCollator.kt

```diff
@@ -60,19 +60,29 @@ internal class DialogSessionUpdate(
  * and [DialogCollator.scheduleUpdates]
  *
  * @param existingSessions [DialogSession] instances that were running before the pending update
+ *
+ * @param onRootUpdateFinished honored only for the outermost [LayeredDialogSessions] caller.
+ * If provided, called with the complete set of [DialogSession] created by
+ * [DialogCollator.scheduleUpdates], not just those of the client identified by [id].
+ * https://github.com/square/workflow-kotlin/issues/1001
```

Comment:
This URL isn't really useful here. It's already in the comments that are at the heart of the full fix.